### ISSUE TITLE
Clicking on minutes resets value for AM/PM and hour

### ISSUE THREAD

**Issue by juhaelee** [arqex/react-datetime#610]
### I'm Submitting a ...

```
[x] Bug report
[ ] Feature request
[ ] Support request
```

**note i am using version 3.0.0-beta.1**

### Steps to Reproduce
Click on the minute arrow and the hour or AM/PM resets to original value. Same when click hour or AM/PM, the minutes reset to original value

![dec-20-2018 19-45-22](https://user-images.githubusercontent.com/606411/50318594-2095ca00-0490-11e9-8eb4-042801882706.gif)

### Expected Results
Minutes to remain the same when clicking hour or AM/PM, vice versa

**Comment by arqex** [comment]
Hey @juhaelee 

Sorry for the delay replying, the end of the year was a difficult time to work on the picker. Can you share how do you use the component, I mean the `<Datetime>` tag with the props?

Thanks for giving a try to the latest version! :)

**Comment by nasrashvilg1** [comment]
was this ever resolved?

**Comment by brianchuchua** [comment]
I ran into this same issue when using this component as primarily a time picker with the following props:

```
dateFormat={false}
timeFormat="HH:mm"
viewMode="time"
timeConstraints={{ minutes: { step: 15 } }}
```

A quick fix I hacked into a fork of this repository was to eliminate this `if` check: https://github.com/YouCanBookMe/react-datetime/blob/master/DateTime.js#L307

`selectedDate` was not being updated consistently whenever a different time was selected, which caused the time picker to remember _either_ the minute selected or the hour selected, but not both. Removing the `if` statement fixed this issue, although it did break the `persistent valid months going monthView->yearView->monthView` test that was present in an older version of this repo.

While the aforementioned hack works for my use case (just using the component as a time picker), there's likely a better fix for the issue, and this approach may very well break the component in other ways.

**Comment by Isarstyle** [comment]
Having the same issue in the version 3.0.0-beta.8.. will this be fixed?

**Comment by e-dong** [comment]
I'm having the same issue in version 3.0.2. I would prefer not to downgrade to 2.16.3 because it is now deprecated and the node-fetch dependency has a denial of service vulnerability: https://www.npmjs.com/advisories/1556

**Comment by arqex** [comment]
Sorry about the delay on addressing this issue. It should be fixed in v3.0.4.
